scripts/compatibility_scanner/compatibility_scanner.q

```diff
@@ -0,0 +1,239 @@
+DEBUG_SHOW_REGEX_PASSED:0b;
+
+ZS_TO_CHECK:(  // .z functions to look out for without the ".z." prefix attached
+  "pm";
+  "zd";
+  "ac";
+  "bm";
+  "exit";
+  "pc";
+  "pd";
+  "pg";
+  "ph";
+  "pi";
+  "po";
+  "pp";
+  "pq";
+  "ps";
+  "pw";
+  "ts";
+  "vs";
+  "wc";
+  "wo";
+  "ws"
+ );
+
+getFullPath:{[path]
+  if[0~count path;:()];
+  if[0h~type path;
+    $[all 10h=type each path;:{x where {not x~()}each x}getFullPath each path;
+      [-2"ERROR: Wrong type for 'path': ",-3!type path;'wrongtype]];
+  ];
+  if[not 10h~type path;-2"ERROR: Wrong type for 'path': ",-3!type path;'wrongtype];
+
+  :@[{first system"readlink -f '",x,"'"};path;
+    {[x;y] -2"ERROR: Could not find path to: '",x,"'";exit 1}[path]];
```

Comment:
y isn't used here, may be useful to show the error?

Reply:
Added it in now